extensions/amp-smartlinks/0.1/link-rewriter/link-rewriter.js

```diff
@@ -157,8 +157,12 @@ export class LinkRewriter {
           resolve();
         });
       });
-
-      chunk(this.rootNode_, task, ChunkPriority.LOW);
+      const elementOrShadowRoot = /** @type {!Element|!ShadowRoot} */ (
+        (this.rootNode_.nodeType == Node.DOCUMENT_NODE)
+          ? this.rootNode_.documentElement
+          : this.rootNode_
+      );
```

Comment:
don't need these outer parens for Js
```
this.rootNode_.nodeType == Node.DOCUMENT_NODE ?
  this.rootNode_.documentElement : this.rootNode_;
```

Reply:
Not my code. This is the vendored stuff I need to use.